### PR TITLE
feat: Added robust dark and light themes and a way to toggle them

### PR DESCRIPTION
**Pull request by Jacksonmills**

### CODE REVIEW

**Comment by adamcohenhillel** [comment]
Looks awesome, a few comments though:

1. text seems a bit off on Iphone 14 max (and tried other sizes too, its the same):
image

File: app/src/components/ThemeToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react"
+import { useTheme } from "next-themes"
+import { Moon, Sun } from "lucide-react";
+import { Button } from "./ui/button";
```
Comment (adamcohenhillel) [inline app/src/components/ThemeToggle.tsx:4]: lets use the "@" import

**Comment by adamcohenhillel** [comment]
The three dots "..." should probably be the same color as the "assistant" bubble, and not the user bubble

image

**Comment by Jacksonmills** [comment]
> Looks awesome, a few comments though:
> 
>     1. text seems a bit off on Iphone 14 max (and tried other sizes too, its the same):
> 
> 
> image

I see it as well! working on a fix now

**Comment by adamcohenhillel** [comment]
Anddd lastly,
I think the light mode can be a bit lighter?

image

Before it had some white and stuff, now it seems a bit darky, what do you think? (we can skip that too)

**Review by adamcohenhillel — APPROVED** [review]

**Comment by Jacksonmills** [comment]
> Looks awesome, a few comments though:
> 
>     1. text seems a bit off on Iphone 14 max (and tried other sizes too, its the same):
> 
> 
> image

Ah so the issue is that firefox handles textarea padding differently, I will go with the lesser of the 2 evils and make it even padding and ff will just have a little bit of a weird initial empty state:
image

**Review by adamcohenhillel — APPROVED** [review]

**Pull request merged**